Helper/PaymentMethods.php

```diff
@@ -474,6 +479,18 @@ protected function getPaymentMethodsRequest(
                 $this->adyenHelper->padShopperReference($this->getCurrentShopperReference());
         }
 
+        $quote = $this->checkoutSession->getQuote();
+
+        $shopperConversionIdData = $quote->getPayment()->getAdditionalInformation('shopper_conversion_id');
+
+        $shopperConversionId = !empty($shopperConversionIdData) ? (string) json_decode($shopperConversionIdData, true) : '';
+
+        if(!empty($shopperConversionId)) {
+            $shopperConversionId = $this->generateShopperConversionId->getShopperConversionId();
+            $paymentMethodRequest["shopperConversionId"] = $shopperConversionId;
+        }
```

Comment:
This block might produce incorrect `shopperConversionId` in the request. Please see the following.

1. There won't be a `shopperConversionId` in the request if the additional information does not have it. However, `getShopperConversionId()` has never been called if there is no additional information. So that, this if block will be skipped forever.
2. Generating a new shopper conversion ID for each `/paymentMethods` call might be a good idea without checking it's existence since the end goal is tracking the correlation between `/paymentMethods` and `/payments` calls.

Apart from that, referring to my previous comments about JSON encoding and decoding to simplify the implementation.